ecs-client.js

```diff
@@ -0,0 +1,102 @@
+'use strict';
+
+const { ECSClient, StopTaskCommand } = require("@aws-sdk/client-ecs");
+const axios = require('axios');
+const callWithRetries = require('./call-with-retries');
+const Exception = require("./exception");
+const exitDeferred = require('./exit-deferred');
+
+const DEFAULT_TIMEOUT = 30000;
+const DEFAULT_REASON = 'ecs-client.js: default stop reason';
+
+class EcsClient {
+
+    /**
+     * @param {Object} options
+     * @param {Object} options.aws - AWS config
+     * @param {String} options.aws.accessKeyId
+     * @param {String} options.aws.secretAccessKey
+     * @param {String} options.aws.region
+     * @param {String} options.metadataUri
+     * @param {Number} [options.timeout=30000] - used for metadata fetch request
+     * @param {Object=} options.logger - An instance of sq-logger
+     */
+    constructor(options) {
+        if(options.logger == null){
+            throw new Exception(Exception.ErrorCode.ERROR_INVALID_PARAMETER, 'An sq-logger instance must be provided.');
+        }
+        this.logger = options.logger;
+        this.metadataUri = options.metadataUri;
+        this.client = new ECSClient(options.aws);
```

Comment:
It's a bit confusing that the difference between this class name and the one from @aws-sdk/client-ecs is the casing.

Reply:
This should never be a problem, since the whole idea of this client class is to provide a more user friendly version of the ECSClient. With this in mind, requiring '@aws-sdk/client-ecs' should never be necessary